pkg/services/ngalert/models/alert_rule.go

```diff
@@ -569,7 +569,7 @@ type AlertRuleVersion struct {
 	Condition       string
 	Data            []AlertQuery
 	IntervalSeconds int64
-	Record          *Record `xorm:"text null 'record'"`
+	Record          *Record `xorm:"json"`
```

Comment:
Manually tested on all 3 supported databases.

As confusing as it seems - this does not mean "use a json column." If the underlying type is TEXT, it'll serialize the value as text. This does not try to map the value to native json types, like postgres's jsonb.